### PR TITLE
Updated default artifacts. Fixes #31.

### LINKED ISSUE #31: Rework default packaging
The default packaging should be changed, as discussed on Slack.

## Current situtation
The build will create 
* a default jar with machine-type classifier, containing only this native library.
* a "uber-jar" with precompiled libraries.

## Critique

* A jar with only one native library is rarely needed.
* Instead, a jar with all the other precompiled libaries, but only one-off/updated (selected arch)  would be much more helpful.
* The "uber-jar" should be the default jar, because it is what developers would expect.
* The warning that an assembly provides the default jar (no classifier) is irritating.
* At the moment, creating fat jars is quite complicated due to this implementation (See #21).

## Proposed Solution

* Replace the default jar with a jar containing all the precompiled libraries.
* Create an additional jar with the existing classifier which includes the one-off (updated) library for the selected system type.

### PR DESCRIPTION
**Pull request by bmhm**
- Default artifact is now the "global" jar containing all precompiled libraries.
  - The system-classifier jar is now an optional jar which is not generated by the package profile.
  - The system-classifier jar contains everything from the "global" jar, but with the system-classifier updated library.
  - Updated cmake output folder to `${cmake.generated.directory}`.
  - Removed MAVEN_OUTPUT_DIRECTORY from CMakeLists.txt.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -258,6 +258,7 @@
         <maven.test.skip>true</maven.test.skip>
         <cmake.generate.skip>true</cmake.generate.skip>
         <cmake.compile.skip>true</cmake.compile.skip>
+        <assembly.one-off-jar.skip>true</assembly.one-off-jar.skip>
```
Comment (tresf) [inline pom.xml]: This can be removed, right?

Comment (bmhm) [reply pom.xml]: YES!

**Comment by tresf** [comment]
As discussed on Slack, the outstanding issue with this PR is that now jars are named incorrectly.  e.g. `m32` and `arm` both come out as `jssc-2.8.1-SNAPSHOT-linux-x86_64-32.jar`.  Once resolved, this can be merged.

File: pom.xml [hunk 2]
```diff
@@ -24,6 +24,7 @@
     <cmake.generate.extra></cmake.generate.extra>
     <cmake.generate.skip>false</cmake.generate.skip>
     <javah.skip>false</javah.skip>
+    <maven.assembly.id>${os.detected.name}-${os.detected.arch}-${os.detected.bitness}</maven.assembly.id>
```
Comment (tresf) [inline pom.xml]: @bmhm there seems to be an odd bug when we use a plugin-generated property such as `${os.detected.name}` from within an assembly.  This bug is not reproducible if we make up our own property.
* One option is to use our own replacement properties throughout
* Another option is to simply declare a single property to be picked up later (That's the one I've chosen)
   * I didn't know if the property name is kosher.  Please rename as needed.

Variable resolution in Maven is unlike any system I've used before.  From what I've seen it does forward lookups , which can be both convenient as well as a pain in the ass.  For example, `<foo.bar>${foo.bar}</foo.bar>` is illegal, which is quite cumbersome as it prevents appending stuff (such as `cmake` flags) to an already existing variable.  I believe this is due to the nature of profiles, but anyway... This above fix seems to work.

**Comment by tresf** [comment]
@bmhm ready for merge once https://github.com/java-native/jssc/pull/32#pullrequestreview-225325149 conversation is resolved.  Just need your advice on the property name that matches convention.  Once you decide this feel free to merge, or I'll do it first thing tomorrow.

File: pom.xml (hunk 2 above)
Comment (bmhm) [reply pom.xml]: That is what I also wanted to propose, so yes, go ahead. We can file an upstream bug, if you wish.

The property name won’t clash. The closest I could find would be `assembly.appendAssemblyId`. So I’ll just mark this as resolved. :)

**Pull request merged**